src/Servers/Kestrel/Core/src/Internal/Http3/Http3Stream.cs

```diff
@@ -821,7 +832,45 @@ private static Task ProcessUnknownFrameAsync()
             await OnEndStreamReceived();
         }
 
-        if (!_isMethodConnect && (_parsedPseudoHeaderFields & _mandatoryRequestPseudoHeaderFields) != _mandatoryRequestPseudoHeaderFields)
+        if (_isProtocolWebTransport)
+        {
+            // Requirement in the spec but this is handled by the Http/3 layer already, so leaving commented out for posterity
+            //if (Scheme != SchemeHttps)
+            //{
+            //    throw new Http3StreamErrorException("Scheme must be HTTPS." /*todo unhardcode*/, Http3ErrorCode.MessageError);
+            //}
+
+            if (!_isMethodConnect)
+            {
+                throw new Http3StreamErrorException("Method must be CONNECT." /*todo unhardcode*/, Http3ErrorCode.ProtocolError); // todo should this be a 400 response instead?
+            }
+
+            if (!ServerOptions.EnableWebTransport)
+            {
+                throw new Http3StreamErrorException("Server has WebTransport disabled. Please enable it prior to setting up WebTransport sessions." /*todo unhardcode*/, Http3ErrorCode.SettingsError);
```

Comment:
Will these strings have to go through some string reviews?

Reply:
Code review is sufficient.

Reply:
If anyone has comments on the strings let me know. I am going to mark this comment as resolved since I moved the strings into `CoreStrings` so this code snippet is now unrelated to the actual text.